indexbuilder.go

```diff
@@ -329,6 +347,38 @@ func (b *IndexBuilder) populateSubRepoIndices() {
 
 const notIndexedMarker = "NOT-INDEXED: "
 
+func (b *IndexBuilder) addSymbols(symbols []*Symbol, doc uint32) error {
+	symStrs := make([]*searchableString, 0, len(symbols))
+	metadataStrs := make([]*searchableString, 0, len(symbols))
+	for _, sym := range symbols {
+		symStr, _, err := b.symbolPostings.newSearchableString([]byte(sym.Name), nil)
+		if err != nil {
+			return err
+		}
+		symStrs = append(symStrs, symStr)
+
+		buf := &bytes.Buffer{}
+		for _, n := range []uint32{doc, sym.Start} {
+			err = binary.Write(buf, binary.BigEndian, n)
+			if err != nil {
+				return nil
+			}
+		}
+		_, err = buf.WriteString(sym.Kind)
+		if err != nil {
+			return err
+		}
+
+		metadataStrs = append(metadataStrs, &searchableString{
+			data: buf.Bytes(),
+		})
+	}
+
+	b.symbolStrings = append(b.symbolStrings, symStrs...)
+	b.symbolMetadata = append(b.symbolMetadata, metadataStrs...)
+	return nil
+}
```

Comment:
can you explain in the PR how symbols are stored in the index. I'm unsure if the current way is the best, I would of expected something that looks more like filename or content (everything concatted together).